bob-backend/src/pearl/group.rs

```diff
@@ -88,6 +90,7 @@ impl Group {
     }
 
     pub async fn remount(&self, pp: impl Hooks) -> AnyResult<()> {
+        let _reinit_lock = self.reinit_lock.write().await;
```

Comment:
`run` should also acquire `reinit_lock`

Reply:
As I understand, reinit_lock is dropped at the end of the function, so it is still acquired while run() is running. Or is it better to drop reinit_lock before run() and reacquire it inside run()?

Reply:
You should refactor `run` function. You can move all the code from `run` function into the new private function (you can name it `run_under_lock` or `run_inner`) and call that function from `remount`. The original `run` function in this case will acquire `reinit_lock` and then call the new private function